[mlir][Affine][NFC] Define AffineForOp operands in ODS

matthias-springer opened the pull request:
Modernize affine dialect ops: Define LB, UB, step and inits as operands in TableGen.

ftynse commented on mlir/include/mlir/Dialect/Affine/IR/AffineOps.td:
```diff
@@ -283,19 +278,17 @@ def AffineForOp : Affine_Op<"for",
     AffineMap getLowerBoundMap() { return getLowerBoundMapAttr().getValue(); }
     AffineMapAttr getLowerBoundMapAttr() {
-      return ::llvm::cast<AffineMapAttr>(*(*this)->getInherentAttr(getLowerBoundAttrStrName()));
+      return ::llvm::cast<AffineMapAttr>(getOperation()->getAttr(getLowerBoundAttrStrName()));
```
I think we could also move the bound maps to listed attributes/properties.

matthias-springer committed:
[mlir][Affine][NFC] Define AffineForOp operands in ODS

Define LB, UB, step and inits as operands in TableGen.

Note: There are still some operands that have not been updated to TableGen yet.
